ostelco-ios-client/ViewControllers/ESim/ESIMPageViewController.swift

```diff
@@ -0,0 +1,169 @@
+            """]
+        case .forMobileDataOnly:
+            return ["""
+            Choose "...for mobile data only"
+            """]
+        case .watchVideo:
+            return ["""
+            Still unsure? Watch this video!
+            """]
+        }
+    }
+    
+    var bottomText: [String]? {
+        switch self {
+        case .instructions:
+            return ["""
+            Open the email on
+            another device
+            """, "another device"]
+        case .scanQRCode:
+            return ["""
+            Scan the QR code
+            """]
+        case .tapToContinue:
+            return nil
+        case .forMobileDataOnly:
+            return nil
+        case .watchVideo:
+            return nil
+        }
+    }
+    
+    var videoURL: URL? {
+        switch self {
+        case .watchVideo:
+            return ExternalLink.esimInstructionsVideo.url
+        default:
+            return nil
+        }
+    }
+    
+    var viewController: ESIMPageViewController {
+        return ESIMPageViewController.fromStoryboard(with: self)
+    }
+}
+
+class ESIMPageViewController: UIViewController {
+
+    @IBOutlet private var topTextLabel: OstelcoLabel!
+    @IBOutlet private var imageView: UIImageView!
+    @IBOutlet private var bottomTextLabel: OstelcoLabel!
+    @IBOutlet private var playerView: UIView!
+    
+    // This is set up in the convenience constructor
+    // swiftlint:disable:next implicitly_unwrapped_optional
+    private(set) var esimPage: ESIMPage!
+    
+    /// Has to be set up through `prepareForSegue` when this VC is loaded
+    // swiftlint:disable:next implicitly_unwrapped_optional
+    private var playerController: AVPlayerViewController!
+    
+    /// Convenience constructor
+    ///
+    /// - Parameter page: The page you wish to display.
+    static func fromStoryboard(with page: ESIMPage) -> ESIMPageViewController {
+        let vc = self.fromStoryboard()
+        vc.esimPage = page
+
+        return vc
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        setTextLabel(topTextLabel, esimPage.topText)
+        
+        if let bottomText = esimPage.bottomText {
+            setTextLabel(bottomTextLabel, bottomText)
+        } else {
+            bottomTextLabel.isHidden = true
+        }
+        
+        if let image = esimPage.image {
+            imageView.image = image
+        } else {
+            imageView.isHidden = true
+        }
+        
+        if let videoURL = esimPage.videoURL {
+            playerController.player = AVPlayer(url: videoURL)
+        } else {
+            playerView.isHidden = true
+        }
+    }
+    
+    private func setTextLabel(_ label: OstelcoLabel, _ parts: [String]) {
+        switch parts.count {
+        case 1:
+            label.text = parts[0]
+        case 2:
+            label.setFullText(parts[0], withBoldedPortion: parts[1])
+        default:
+            break
```

Comment:
I would really, really strongly recommend separating these into `text` and optional `boldedText` options rather than passing them as an array - you don't need to have arbitrary array of things to be bolded, just the binary Is Anything Bolded Y/N

Reply:
@designatednerd I agree, it felt a bit hacky when I did it, not sure quite sure how to change the logic to use a binary to check if anything is bolded or not.
My initial thought was to return an Attributed String directly, then at least I wouldn't need to return multiple values from each variable that can have a bolded part. Or I could have a separate variable containing the boldedPortion for both top and bottom text.

I could also use a tuple:
```
var bottomText: (String?, String?) { <-- could also be named tuple?
        switch self {
        case .instructions:
            return ("""
            Open the email on
            another device
            """, "another device")
       .... all the other cases ...
       }
}
```

and then set it as follows:

```
let (bottomText, boldedPortion) = esimPage.bottomText
if let bottomText = bottomText {
          if let boldedPortion = boldedPortion {
                  bottomTextLabel.setFullText(bottomText, withBoldedPortion: boldedPortion)
          } else {
                  bottomTextLabel.text = bottomText
          }
}
```

Or create an extra variable to hold optional bolded text for both top and bottom:

```
var bottomText: String? {
        switch self {
        ...
        }
}
var bottomTextBoldedPortion: String? {
        switch self {
        ...
        }
}
```

and set it as follows:

```
if let bottomText = esimPage.bottomText {
        if let boldedPortion = esimPage.bottomTextBoldedPortion {
                bottomTextLabel.setFullText(bottomText, withBoldedPortion: boldedPortion)
        } else {
                bottomTextLabel.text = bottomText
        }
}
```

Reply:
After looking at #231 I think I get what you where talking about, if I change the return value to a struct which contains text and optional bolded text then I could add a function in the struct to check if it has a bolded part

Reply:
That could work although it's probably easier in the end to just `if let` the bolded portion when you're looking at it since you're going to need to unwrap that anyway